dual-callback.ts

```diff
@@ -31,6 +31,7 @@ export class CancellableDualCallbackProducer<T> implements Producer<T> {
                  private cancel: (cancelId: number) => void,
                  ...rest: any[]) {
         this.rest = rest
+        this.cancelId = -1
```

Comment:
`-1` seems the wrong thing to initialize to without knowing more about individual functions this is wrapping and their valid id restrictions. If it needs to be initialized I'd probably push towards using a `number | null` field instead and then wrap the cancel logic itself with an `if (this.cancelId !== null) {` for safety.